util/network-devp2p/src/discovery.rs

```diff
@@ -537,6 +549,7 @@ impl Discovery {
 
 		response.append(&echo_hash);
 		append_expiration(&mut response);
+		response.append(&self.enr.seq());
```

Comment:
Why can't we use `enr_seq` (`val_at(4)` in the `PING` packet) to determine if we should include `enr` in the response or not?
That seems to me like a change that would maintain backward compatibility (i.e. if we receive `enr_seq` in `PING` we respond in `PONG`), as oppossed to always including it in the response.

Reply:
Done